### PR TITLE
Update how to connect TiDB by MySQL Client and the limitations of the global indexes

### PR DESCRIPTION
**Pull request by Frank945946**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Update how to connect TiDB by MySQL Client and the limits of the global indexes 

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.5 (TiDB 8.5 versions)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-connect-to-tidb.md [hunk 1]
```diff
@@ -29,6 +29,18 @@ sudo yum install mysql
...
+```shell
+brew install mysql-client@8.0
+brew unlink mysql
+brew link mysql-client@8.0
+```
+如果还是遇到问题，可以尝试指定 MySQL v8.0 Client 的安装路径来使用 MySQL v8.0 Client 连接 TiDB。
+```shell
+/opt/homebrew/opt/mysql-client@8.0/bin/mysql --comments --host [YOUR_IP_ADDRESS] --port [YOUR_PORT_NUMBER] -u your_user_name -p
+```
+请使用实际部署的 MySQL v8.0 Client 替代上述的路径。
```
Comment (hfxsd) [inline develop/dev-guide-connect-to-tidb.md:43]:
```suggestion

macOS 的 MySQL v9.0 Client 无法正确加载 `mysql_native_password` 插件，导致连接 TiDB 时报错 `ERROR 2059 (HY000): Authentication plugin 'mysql_native_password' cannot be loaded`。为解决该问题，建议安装并使用 MySQL v8.0 Client 链接 TiDB 。安装命令如下：
 
```shell
brew install mysql-client@8.0
brew unlink mysql
brew link mysql-client@8.0
```

如果仍然遇到问题，可以尝试指定 MySQL v8.0 Client 的安装路径来使用 MySQL v8.0 Client 连接 TiDB。安装命令如下：

```shell
/opt/homebrew/opt/mysql-client@8.0/bin/mysql --comments --host ${YOUR_IP_ADDRESS} --port ${YOUR_PORT_NUMBER} -u ${your_user_name} -p
```
```

File: partitioned-table.md [hunk 2]
```diff
@@ -1775,7 +1775,9 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
 
 - 如果索引定义中未显式指定 `GLOBAL` 关键字，TiDB 将默认创建局部索引 (Local Index)。
 - `GLOBAL` 和 `LOCAL` 关键字仅适用于分区表，对非分区表没有影响。即在非分区表中，全局索引和局部索引之间没有区别。
-- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`、`SPLIT PARTITION` 和 `EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`EXCHANGE PARTITION`、`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
+- 当前仅支持创建 Unique Global Index，如果需要对非 Unique 的列创建 Global Index，该 Global Index 需要包含分区建形成复合索引，且非 Unique 的列排在前面。如非 Unique 的列为 col3，分区建为 col1，则可以通过 “alter table xxx add unique key(col3,col1) global;”，实现对 非 Unique 列 col3 创建 Global Index。
+- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`，`EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
+- 创建了 Global Index 的表不支持 `EXCHANGE PARTITION`。
```
Comment (hfxsd) [inline partitioned-table.md:1780]:
```suggestion
- 当前仅支持创建 Unique 全局索引，如果需要对非 Unique 的列创建全局索引，该全局索引需要包含分区建形成复合索引，且非 Unique 的列排在前面。例如，非 Unique 的列为 `col3`，分区建为 `col1`，则可以通过执行 `ALTER TABLE ... ADD UNIQUE KEY(col3,col1) GLOBAL;`，实现对非 Unique 列 `col3` 创建全局索引。
- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`，`EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
- 创建了全局索引的表不支持 `EXCHANGE PARTITION`。
```

File: partitioned-table.md [hunk 3]
```diff
@@ -1775,7 +1775,9 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
 
 - 如果索引定义中未显式指定 `GLOBAL` 关键字，TiDB 将默认创建局部索引 (Local Index)。
 - `GLOBAL` 和 `LOCAL` 关键字仅适用于分区表，对非分区表没有影响。即在非分区表中，全局索引和局部索引之间没有区别。
-- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`、`SPLIT PARTITION` 和 `EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`EXCHANGE PARTITION`、`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
+- 当前仅支持创建唯一全局索引 (Unique Global Index)，如果需要对非唯一的列创建全局索引，该全局索引需要包含分区建形成复合索引，且非唯一的列排在前面。例如，非唯一的列为 `col3`，分区建为 `col1`，则可以通过执行 `ALTER TABLE ... ADD UNIQUE KEY(col3,col1) GLOBAL;` 为非唯一列 `col3` 创建全局索引。
+- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`，`EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
+- 创建了全局索引的表不支持 `EXCHANGE PARTITION`。
```
Comment (hfxsd) [inline partitioned-table.md:1780]:
```suggestion
- 当前仅支持为唯一列创建全局索引 (Unique Global Index)。如果需要对非唯一列创建全局索引，该索引必须包含分区键并形成复合索引，且非唯一列应位于分区键之前。例如，如果非唯一列是 `col3` 而分区键是 `col1`，可以通过执行以下SQL语句为 `col3` 创建全局索引：
  
    ```sql
    ALTER TABLE ... ADD UNIQUE INDEX(col3, col1) GLOBAL;
    ```

- 以下 DDL 操作会触发全局索引的更新：`ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION` 和 `EXCHANGE PARTITION`。这些 DDL 操作将在全局索引更新完成后才返回结果。因此，这可能会延长通常需要快速完成的 DDL 操作的执行时间，例如数据归档操作（`TRUNCATE PARTITION` 和 `DROP PARTITION`）。如果没有全局索引，这些 DDL 操作可以立即执行完成。
- 包含全局索引的表不支持 `EXCHANGE PARTITION`。
```

File: develop/dev-guide-connect-to-tidb.md [hunk 4]
```diff
@@ -30,6 +30,20 @@ sudo yum install mysql
 mysql --host <tidb_server_host> --port 4000 -u root -p --comments
 ```
 
+macOS 的 MySQL v9.0 Client 无法正确加载 `mysql_native_password` 插件，导致连接 TiDB 时报错 `ERROR 2059 (HY000): Authentication plugin 'mysql_native_password' cannot be loaded`。为解决该问题，建议安装并使用 MySQL v8.0 Client 链接 TiDB 。安装命令如下：
```
Comment (lilin90) [inline develop/dev-guide-connect-to-tidb.md:33]:
```suggestion
macOS 上的 MySQL v9.0 客户端无法正确加载 `mysql_native_password` 插件，导致连接 TiDB 时报错 `ERROR 2059 (HY000): Authentication plugin 'mysql_native_password' cannot be loaded`。为解决该问题，建议安装并使用 MySQL v8.0 客户端来连接 TiDB 。安装命令如下：
```

File: develop/dev-guide-connect-to-tidb.md [hunk 5]
```diff
@@ -30,6 +30,20 @@ sudo yum install mysql
...
+ 
+```shell
+brew install mysql-client@8.0
+brew unlink mysql
+brew link mysql-client@8.0
+```
+
+如果仍然遇到问题，可以尝试指定 MySQL v8.0 Client 的安装路径来使用 MySQL v8.0 Client 连接 TiDB。安装命令如下：
```
Comment (lilin90) [inline develop/dev-guide-connect-to-tidb.md:41]:
```suggestion
如果仍然遇到问题，可以尝试指定 MySQL v8.0 客户端的安装路径来使用 MySQL v8.0 客户端连接 TiDB。连接命令如下：
```

File: partitioned-table.md [hunk 6]
```diff
@@ -1775,7 +1775,14 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
 
 - 如果索引定义中未显式指定 `GLOBAL` 关键字，TiDB 将默认创建局部索引 (Local Index)。
 - `GLOBAL` 和 `LOCAL` 关键字仅适用于分区表，对非分区表没有影响。即在非分区表中，全局索引和局部索引之间没有区别。
-- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`、`SPLIT PARTITION` 和 `EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`EXCHANGE PARTITION`、`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
+- 当前仅支持为唯一列创建全局索引 (Unique Global Index)。如果需要对非唯一列创建全局索引，该索引必须包含分区键并形成复合索引，且非唯一列应位于分区键之前。例如，如果非唯一列是 `col3` 而分区键是 `col1`，可以通过执行以下SQL语句为 `col3` 创建全局索引：
```
Comment (lilin90) [inline partitioned-table.md:1778]:
```suggestion
- 当前仅支持为唯一列创建全局索引 (Unique Global Index)。如果需要对非唯一列创建全局索引，该索引必须包含分区键并形成复合索引，且非唯一列应位于分区键之前。例如，如果非唯一列是 `col3` 而分区键是 `col1`，可以通过执行以下 SQL 语句为 `col3` 创建全局索引：
```

File: partitioned-table.md [hunk 7]
```diff
@@ -1775,7 +1775,14 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
...
+  
+    ```sql
+    ALTER TABLE ... ADD UNIQUE INDEX(col3, col1) GLOBAL;
+    ```
+
+- 以下 DDL 操作会触发全局索引的更新：`ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION` 和 `EXCHANGE PARTITION`。这些 DDL 操作将在全局索引更新完成后才返回结果。因此，这可能会延长通常需要快速完成的 DDL 操作的执行时间，例如数据归档操作（`TRUNCATE PARTITION` 和 `DROP PARTITION`）。如果没有全局索引，这些 DDL 操作可以立即执行完成。
```
Comment (lilin90) [inline partitioned-table.md:1784]: 调整了下措辞使其更顺畅，请确认下意思 @Frank945946 
```suggestion
- 以下 DDL 操作会触发全局索引的更新：`ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION` 和 `EXCHANGE PARTITION`。这些 DDL 操作将在全局索引更新完成后才返回结果，因此可能会延长执行时间，尤其是对于通常需要快速完成的 DDL 操作，例如数据归档操作（`TRUNCATE PARTITION` 和 `DROP PARTITION`）。如果没有全局索引，这些 DDL 操作可以立即完成。
```

**Review by Icemap — APPROVED** [review]
LGTM, and don't forget the English edition.

File: develop/dev-guide-connect-to-tidb.md [hunk 8]
```diff
@@ -30,6 +30,20 @@ sudo yum install mysql
...
+brew unlink mysql
+brew link mysql-client@8.0
+```
+
+如果仍然遇到问题，可以尝试指定 MySQL v8.0 客户端的安装路径来使用 MySQL v8.0 客户端连接 TiDB。连接命令如下：
+
+```shell
+/opt/homebrew/opt/mysql-client@8.0/bin/mysql --comments --host ${YOUR_IP_ADDRESS} --port ${YOUR_PORT_NUMBER} -u ${your_user_name} -p
+```
+
```
Comment (hfxsd) [inline develop/dev-guide-connect-to-tidb.md:46]:
```suggestion

请使用实际部署的 MySQL v8.0 客户端的安装路径替代上述命令中的 `/opt/homebrew/opt/mysql-client@8.0/bin/mysql`。
```

File: partitioned-table.md [hunk 9]
```diff
@@ -1775,7 +1775,14 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
...
 - 如果索引定义中未显式指定 `GLOBAL` 关键字，TiDB 将默认创建局部索引 (Local Index)。
 - `GLOBAL` 和 `LOCAL` 关键字仅适用于分区表，对非分区表没有影响。即在非分区表中，全局索引和局部索引之间没有区别。
-- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`、`SPLIT PARTITION` 和 `EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`EXCHANGE PARTITION`、`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
+- 当前仅支持为唯一列创建全局索引 (Unique Global Index)。如果需要对非唯一列创建全局索引，该索引必须包含分区键并形成复合索引，且非唯一列应位于分区键之前。例如，如果非唯一列是 `col3` 而分区键是 `col1`，可以通过执行以下 SQL 语句为 `col3` 创建全局索引：
+  
+    ```sql
+    ALTER TABLE ... ADD UNIQUE INDEX(col3, col1) GLOBAL;
+    ```
+
+- 以下 DDL 操作会触发全局索引的更新：`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION` 和 `EXCHANGE PARTITION`。这些 DDL 需等待全局索引更新完成后才会返回结果，耗时会相应增加。尤其是在数据归档场景下，如 `DROP PARTITION` 和 `TRUNCATE PARTITION`，若没有全局索引，通常可以立即完成；但使用全局索引后，耗时会随着所需更新的索引数量而增加。
```
Comment (hfxsd) [inline partitioned-table.md:1784]:
```suggestion
- 以下 DDL 操作会触发全局索引的更新：`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION` 和 `EXCHANGE PARTITION`。这些 DDL 需等待全局索引更新完成后才会返回结果，耗时会相应增加。尤其是在数据归档场景下，如 `DROP PARTITION` 和 `TRUNCATE PARTITION`，若没有全局索引，通常可以立即完成；但使用全局索引后，耗时会随着所需更新的索引数量的增加而增加。
```

File: partitioned-table.md [hunk 10]
```diff
@@ -1775,7 +1775,14 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
 
 - 如果索引定义中未显式指定 `GLOBAL` 关键字，TiDB 将默认创建局部索引 (Local Index)。
 - `GLOBAL` 和 `LOCAL` 关键字仅适用于分区表，对非分区表没有影响。即在非分区表中，全局索引和局部索引之间没有区别。
-- DDL 操作如 `ADD PARTITION`、`DROP PARTITION`、`TRUNCATE PARTITION`、`REORGANIZE PARTITION`、`SPLIT PARTITION` 和 `EXCHANGE PARTITION` 等也会触发对全局索引的更新，这些 DDL 的执行结果将在全局索引更新完成后才会返回。因此，这可能会延迟一些通常需要快速完成的 DDL 的操作，如数据归档操作（`EXCHANGE PARTITION`、`TRUNCATE PARTITION` 和 `DROP PARTITION`）。而如果没有全局索引，这些 DDL 操作可以立即执行完成。
+- 当前仅支持为唯一列创建全局索引 (Unique Global Index)。如果需要对非唯一列创建全局索引，该索引必须包含分区键并形成复合索引，且非唯一列应位于分区键之前。例如，如果非唯一列是 `col3` 而分区键是 `col1`，可以通过执行以下 SQL 语句为 `col3` 创建全局索引：
```
Comment (Defined2014) [inline partitioned-table.md:1778]:
```suggestion
- 当前仅支持为唯一列创建全局索引 (Unique Global Index)。如果需要对非唯一列创建全局索引，可以通过包含主键并形成复合索引。例如，如果非唯一列是 `col3` 而主键是 `col1`，可以通过执行以下 SQL 语句为 `col3` 创建全局索引：
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by hfxsd — APPROVED** [review]

**Pull request merged**